litmus-portal/frontend/src/pages/AnalyticsPage/index.tsx

```diff
@@ -25,17 +78,166 @@ const AnalyticsPage: React.FC = () => {
     { variables: { projectID: selectedProjectID } }
   );
 
-  const workflow = data?.getWorkFlowRuns.filter(
-    (w) => w.workflow_run_id === workflowRunId
-  )[0];
+  const [workflowRunDataForPlot, setWorkflowRunDataForPlot] = React.useState<
+    WorkflowRunData[]
+  >([]);
+
+  useEffect(() => {
+    const workflowRuns: WorkflowRunData[] = [];
+
+    let testDate: string = '';
+
+    const selectedWorkflows = data?.getWorkFlowRuns.filter(
+      (w) => w.workflow_run_id === workflowRunId
+    );
+
+    const workflowRunsExecutionData: ExecutionData[] = [];
+    selectedWorkflows?.forEach((data) => {
+      try {
+        const executionData = JSON.parse(data.execution_data);
```

Comment:
```suggestion
        const executionData = JSON.parse(data.execution_data as ExecutionData);
```

Reply:
actually its a string here @arkajyotiMukherjee

Reply:
I know...that why when you parse it...you can type it is ExecutionData so that you can access all it's properties...This has been done many time in variables pages....take a look at WorkflowDetails page